### PR TITLE
Add epsilon after precision-factor-multiplication

### PR DESCRIPTION
**Pull request by michaeldeistler**
Fixes [this issue of SNPE-C](https://github.com/mackelab/sbi/issues/669).

The change is that we apply the epsilon addition after having multiplied the precision factors. I.e., instead of
```
P = (L+eps) (L+eps)^T
```
we now do
```
P = (L L^T) + eps
```

This fixes SNPE-C on the toy example posted in the issue posted above.

I also decreased the magnitude of `eps` because previously, it was essentially squared through the `(L+eps) (L+eps)^T` operation.

### CODE REVIEW

**Review by janfb — APPROVED** [review]
makes sense, thanks!

File: pyknos/mdn/mdn.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def __init__(
             raise NotImplementedError
 
         # Constant for numerical stability.
-        self._epsilon = 1e-2
+        self._epsilon = 1e-4
```
Comment (janfb) [inline pyknos/mdn/mdn.py:84]: 👍

Comment (janfb) [reply pyknos/mdn/mdn.py:84]: we should add a test in `sbi` to make sure this really fixes the issue.

Comment (michaeldeistler) [reply pyknos/mdn/mdn.py:84]: Agreed, I'll create an issue

**Pull request merged**